### ISSUE TITLE
[feat] <rh-table>

### ISSUE THREAD

**Issue by ronniemcbride** [RedHat-UX/red-hat-design-system#431]
### Description

This component must accommodate the need for a standardized table data style for all digital experiences. This component exists in the Patternfly Design System, but the purpose was to support large data sets. This component will support large data sets and multiline content types.

### Acceptance Criteria

- [x] Design done
- [ ] Development done

### Image

![table](https://user-images.githubusercontent.com/73180910/179006337-f4ecf388-3ada-48d0-ba43-84b1f7e4e695.png)

### Link to design doc

https://xd.adobe.com/view/a598a299-20af-4a97-b37e-cbc04931e112-f0cb/screen/0ac8b4cf-f544-4cfb-8a2e-7600d5db09ee/

### Other resources

https://docs.google.com/document/d/1BHsSVS99DMmTUqEKFTRz7pPLB2ZhcNF1fzOOhCIfqPQ/edit#heading=h.nui944fdk774

**Comment by zeroedin** [comment]
Just to follow up on chat here:

https://www.npmjs.com/package/@cpelements/rh-table

A version of `rh-table` was developed back in January by Rob Lahoda.   

I don't remember the specifics but I believe the intent was to migrate this to Lit and into RHDS when our team got there with the project, at the time the customer portal team had already had prior art on tables, so we asked for modifications to work with our "design".    I believe this to be the l live code on RHDC.   It is currently written in PFE v1.2 style framework.

Comments from the source code:
```
// @todo See if there are more a11y features we should take from: https://dequeuniversity.com/library/aria/table-sortable
// @todo See if <dialog> tag could be useful for the maximized table
// @todo Bug where fullscreen button doesn't show up sometimes in smaller reflow (hard to replicate)
// @todo Pagination?
// @todo Search?
// @todo move to pfe-tools/pfe 2.0
// @todo after pfe 2.0 write tests
//    - Test to make sure full screen button shows up when table scrolls
//    - Tests for sorting results
```

I have a question out to both Rob and Wes Ruvalcaba about its status.

**Comment by zeroedin** [comment]
After a conversation with Wes, we are good to go to migrate it from its current state to RHDS and lit.

**Comment by wesruv** [comment]
TL;DR
I would love to kill CP Elements and everything in it in favor of anything based on Lit element. Unfortunately I'm not sure I can actually run any non PFE 1.x components on the portal without a giant headache, so we continued development in cp-elements.

It looks like our design evolved from the design in this ticket, we have mockup for sorting, the full screen capability (which I know not everyone wants so maybe it doesn't go in this component?)

I can do a demo of what we have, or folks can take a look by cloning, nvm use, installing, and running npm start on cp-elements.

![image](https://user-images.githubusercontent.com/5607236/179057017-c2afd505-7641-415b-b916-093bfb1965be.png)
![image](https://user-images.githubusercontent.com/5607236/179057053-f646d1fd-a083-45a1-906c-1ea019bece4d.png)
![image](https://user-images.githubusercontent.com/5607236/179057083-1d9f5510-6505-4363-a116-83e018a5070b.png)

Maximize table available: 
![image](https://user-images.githubusercontent.com/5607236/179057194-1b73e089-1fd5-49f4-aabe-049f68fad6d7.png)

Maximized:
![image](https://user-images.githubusercontent.com/5607236/179057299-4a66eaca-01d7-4c72-a230-c041133d0903.png)

**Comment by bennypowers** [comment]
moved back to "design needed" per @ronniemcbride 's comment

**Comment by markcaron** [comment]
@wesruv can we move forward with finishing development on the original set of features / MVP, and move these enhancements into a new feature request? This way we can get the table out quicker to folks.

**Comment by marionnegp** [comment]
Live examples of the intersection hover and maximize table features: [https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.6/html/install/installing#network-configuration-infra-operator](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.6/html/install/installing#network-configuration-infra-operator)

**Comment by marionnegp** [comment]
Live examples of the intersection hover and maximize table features: [https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.6/html/install/installing#network-configuration-infra-operator](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.6/html/install/installing#network-configuration-infra-operator)

**Comment by eyevana** [comment]
Guess what lucky gal is picking this up? :P @wesruv @markcaron which features are MVP and which ones can be broken off into follow-up tasks?

**Comment by eyevana** [comment]
@coreyvickery @markcaron I also received these [Figma designs](https://www.figma.com/file/55pIOeQu2ry4oaGvhrkLjM/AsciiDoc-Elements?type=design&node-id=0-1&mode=design&t=rM0lR94OoaSnR6AV-0) from Adela, I'm not sure what designs to follow at this point. Can we work on the docs for rh-table and consolidate design differences? Here's my [WIP](https://deploy-preview-1197--red-hat-design-system.netlify.app/elements/table/demo/).

**Comment by coreyvickery** [comment]
@eyevana Use Adela's.
